src/main/java/org/cryptomator/common/vaults/Vault.java

```diff
@@ -375,6 +364,15 @@ public String getEffectiveMountFlags() {
 		}
 	}
 
+	public UnverifiedVaultConfig getUnverifiedVaultConfig() {
+		try {
+			return configWrapper.getConfig();
+		} catch (IOException e) {
+			throw new IllegalStateException("Vault Config not present.");
+		}
```

Comment:
> I'd keep a checked exception, so we can react to it accordingly. 

Sigh, how about i implement a second `getConfigSafe()` method?

Because from a theoretical point of view, classes using this method _should_ never trigger the exception. If the config is not present, the vault _should_ already be in the error state and hence the UI should prohibit any actions involving it. (i know a lotta shoulds) Due to the unfitting architecture, i can't guarantee it, though.

Reply:
"The other way round will it become a shoe": If there are places in the code where we can assume the method not to throw, it should be named "getUnchecked" but the default "get" should throw. That way the method name suggests that the developer is knowingly accepting the risks.

Once we're done with this, we need a major refactoring though. The finite automata suggestion would solve this, as an "VaultInError" would just not offer such a method.